### PR TITLE
Add mathjax support

### LINKED ISSUE #164: MathJaX v3 support
Hi there, I am author of the following plugins

- MathJaX v3 plugin of VNote https://github.com/tamlok/vnote/pull/1221
- MathJaX v3 plugin of Remark https://github.com/remarkjs/remark-math/pull/40
- MathJaX v3 plugin of Markdown-It https://github.com/nzt/markdown-it-mathjax3

As I'd like to use MathJaX v3 in map-core, I open this issue. MathJaX v3 has a new plugin for bussproofs missing in the older versions, and also, has a SVG rendering without any fonts. There is good conformance for your project at the SVG feature.

 I've already implemented markdown-it plugin with TypeScript based on markdown-it-katex as like marp-core. Thus, I can add MathJaX plugin to your project immediately.

I'd like to ask you, which options do you choose?
1. Replacing KaTeX plugin with MathJaX plugin
2. Adding MathJaX plugin, and keeping KaTeX plugin

Thanks!

### PR DESCRIPTION
**Pull request by nzt**
Closes #164

### CODE REVIEW

File: src/marp.ts [hunk 1]
```diff
@@ -25,6 +26,7 @@ export interface MarpOptions extends Options {
           | { [attr: string]: boolean | ((value: string) => string) }
       }
   markdown?: object
+  mathjax?: boolean
```
Comment (yhatt) [inline src/marp.ts:29]: I think that interface of constructor option is not intuitive. `math: 'mathjax'` would be better than an individual option.

File: src/mathjax/math.ts [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+      },
+      { alt: ['paragraph', 'reference', 'blockquote', 'list'] }
+    )
+
+    md.renderer.rules.marp_math_block = (tokens, idx) => {
+      const { content } = tokens[idx]
+
+      try {
+        return `<p>${adaptor.outerHTML(
+          mathDocument.convert(content, genOpts(true))
```
Comment (yhatt) [inline src/mathjax/math.ts:96]: By comparing to KaTeX rendering, the rendered math block is not centering.

Comment (nzt) [reply src/mathjax/math.ts:96]: Solved, check it.

```js
const { Marp } = require("./lib/marp.js")
const marp = new Marp({ math: true, mathjax: false})

const md = `
# Elementary Math

$$
1+1 = 2
$$
`
const r = marp.render(md)
console.log(r.html + `<style>${r.css}</style>`)
```

File: src/marp.ts (hunk 1 above)
Comment (nzt) [reply src/marp.ts:29]: Nice idea. Let me confirm you.

```
math: 'katex' # enable katex as same as `true`
math: true # enable katex for backward compatibility
math: { ... } # enable katex with options
math: 'mathjax' # enable mathjax
```

Do you want to do like this?

**Comment by nzt** [comment]
My updates have been done by #166 . Thus, I stop to update this branch.

**Pull request closed without merging**